test/integration/integration_test.go

```diff
@@ -505,6 +506,13 @@ func (i *integrationTest) exerciseIncrementalBuild(tag, imageName string, remove
 	if checkOnBuild {
 		i.fileExists(containerID, "/sti-fake/src/onbuild")
 	}
+
+	if took := time.Since(start); took > docker.DefaultDockerTimeout {
+		// https://github.com/openshift/source-to-image/issues/301 is a
+		// case where incremental builds would get stuck until the
+		// timeout.
+		t.Errorf("Test took too long (%v), some operation may have gotten stuck waiting for the DefaultDockerTimeout (%v). Inspect the logs to find operations that took long.", took, docker.DefaultDockerTimeout)
+	}
```

Comment:
There's also a sort-of corollary test which would be nice to have: implement the semi-pathological s2i build, which at every step carries out actions which run for longer than DefaultDockerTimeout, which should always succeed.  Don't know whether you consider that within scope for this change, or whether I should open an issue for that?